src/sinks/honeycomb.rs

```diff
@@ -65,6 +71,34 @@ fn default_endpoint() -> String {
     "https://api.honeycomb.io/1/batch".to_string()
 }
 
+/// Configures how the sample rate is determined for each event. If an integer sample rate is found on the
+/// event from the configured source, it will be removed from the event body and used as the sample rate for
+/// that event for Honeycomb to inflate query results. If the sample rate source is disabled, no fields will
+/// be removed and no sample rate will be sent to Honeycomb.
```

Comment:
This is a good description and I think it could be moved up to the actual config member so that it's used in the auto generated docs.